src/pybind/mgr/devicehealth/module.py

```diff
@@ -220,6 +221,44 @@ def config_notify(self):
                     self.get_module_option(opt['name']))
             self.log.debug(' %s = %s', opt['name'], getattr(self, opt['name']))
 
+    def notify(self, notify_type, notify_id):
+       # create device_health_metrics pool if it doesn't exist
+       if notify_type == "osd_map":
+            if not self.device_pool:
+                self.create_pool()
```

Comment:
```suggestion
        if self.pool_name not in self.rados.list_pools():
            self.create_pool()
```

Reply:
Recurrent! `create_pool` does the same check.